bluesky/callbacks/core.py

```diff
@@ -42,7 +42,13 @@ def f(*args, **kwargs):
 class CallbackBase:
     def __call__(self, name, doc):
         "Dispatch to methods expecting particular doc types."
-        return getattr(self, name)(doc)
+        # if callback returns it will return a new dict document
+        ret = getattr(self, name)(doc)
+        if ret:
```

Comment:
This `if` statement raises a red flag for me. I would like to `return name, getattr(self, name)(doc)` in all cases. If a callback was not designed for a pipeline (and thus returns `None`) it should be up to the caller to say, "OK then pass through the original `doc`."

Reply:
I agree with @danielballan from an explicitness PoV.  But I also agree with @CJ-Wright from a 'gentle migration PoV' this is a reasonable path forward.

Maybe

```python
ret = getattr(self, name)(doc)
return name, ret if ret is not None else doc
```

Reply:
Is that smoother change from the current behavior? How so?

Reply:
Is that smoother change from the current behavior? How so?

Reply:
This is an API change no matter what (as now we are returning a tuple with `name` rather than what ever was happening before), however doing "the right thing" if the user returns `None` means that we do not have to go through and add dummy `return doc` to every callback (the 'gentle migration' is on the produces side).  This is a pattern we use in caproto too, returning `None` means "no opinion" letting the outer function have opinions on what to do next (or conversely letting the inner function not have opinions about what to do next).

Testing on `None` seems more defensive than testing on truth.  We don't have any valid documents that are currently 'falsy' that may not always be true (a 0 row page?) or maybe we want to let non-document things flow through this (should `stop` be allowed to return fit values?).

If some pipeline wants a 'stop propogating' flag it can either use `False` or it's own sentinel. 

On the other-other hand, maybe we want to have that discussion here?

Reply:
To avoid breakage, let's leave this alone and make a new thing, which we can use in the callbacks rewrite.

Reply:
There is more than one good reason to do this. See https://github.com/NSLS-II/event-model/pull/29